libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.ts

```diff
@@ -0,0 +1,86 @@
+import {
+    Component,
+    ContentChildren,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    EventEmitter,
+    Input,
+    Optional,
+    Output,
+    QueryList,
+    Self,
+    ViewEncapsulation,
+    ViewChildren
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { CollectionBaseInput } from '../collection-base.input';
+import { CheckboxComponent } from '../checkbox/checkbox.component';
+import { PlatformCheckboxChange } from '../checkbox/checkbox.component';
+import { FormFieldControl } from '../form-control';
+
+/**
+ * Checkbox group implementation based on the
+ * https://github.com/SAP/fundamental-ngx/wiki/Platform:-CheckboxGroup-Technical-Design
+ * documents.
+ *
+ */
+
+@Component({
+    selector: 'fdp-checkbox-group',
+    templateUrl: './checkbox-group.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [{ provide: FormFieldControl, useExisting: CheckboxGroupComponent, multi: true }]
+})
+export class CheckboxGroupComponent extends CollectionBaseInput {
+    /**
+     * value for selected checkboxes.
+     */
+    get value(): any {
+        return super.getValue();
+    }
+    set value(selectedValue: any) {
+        super.setValue(selectedValue);
+    }
+
+    /**
+     * To Dispaly multiple checkboxes in a line
+     */
+    @Input()
+    isInline: boolean = false;
+
+    /** Children checkboxes passed as content */
+    @ContentChildren(CheckboxComponent)
+    contentCheckboxes: QueryList<CheckboxComponent>;
+
+    /** Children checkboxes created from passed list values. */
+    @ViewChildren(CheckboxComponent)
+    viewCheckboxes: QueryList<CheckboxComponent>;
+
+    @Output()
+    readonly valueChange: EventEmitter<PlatformCheckboxChange> = new EventEmitter<PlatformCheckboxChange>();
+
+    constructor(
+        private _changeDetector: ChangeDetectorRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm
+    ) {
+        super(_changeDetector, ngControl, ngForm);
+    }
+
+    writeValue(value: any): void {
+        if (value) {
+            super.writeValue(value);
+        }
+        this._changeDetector.detectChanges();
+    }
+
+    /**
+     * raises event when Checkbox group value changes.
+     * @param event: contains checkbox and event in a PlatformCheckboxChange class object.
+     */
+    public groupValueChanges(event: PlatformCheckboxChange): void {
+        this.onTouched();
```

Comment:
Why did you decided to call OnTouched here on the onChange ?

Reply:
error will only trigger when control is touched. By just changing the group value, touched property is not getting set.
so whenever change event is triggered, i am touching the control. so when the value is not there for required field, it will set error.

Is there any other way to do this?